### DISCUSSION TITLE
how to properly us a *file* dependency that isn't a zip file inside of a port?

### DISCUSSION THREAD

**Discussion by Cazadorro** [microsoft/vcpkg#36039] [Q&A]
I have a non extractable file that needs to be used as a port's dependency, that I would normally get through fetchcontent. My problem is that unless I use one of VCPKG's blessed functions, it looks like I can't easily have a `"${SOURCE_PATH}"` to use for my  files location.  I have a CMakeLists.txt and a /cmake directory I intended to copy into a valid `${SOURCE_PATH}`, which I hoped would enable me to have VCPKG pretend it was a regular cmake project download. 

I was looking at `vcpkg_download_distfile` but it appears to pretty much [always be paired](https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_extract_source_archive) with a "extract_source_archive" or something which retrieves the source location, so I'm not sure how to manage that myself. 

How do I handle a single file dependency (not a whole repo, or cmake project) that I need to download, and properly use a ${SOURCE_PATH} for my port?

**Comment by dg0yt** [comment]
(With the background of #36037, but in more general words:)

`SOURCE_PATH` is just a variable pointing to the project source. If the build rules must be maintained in the port, the variable can be set to a subdir of the port dir.
- Offer `CMakeList.txt` and any immutable auxiliary files, e.g. C sources, in this `SOURCE_PATH` dir.
- Pass the download location of the desired single file as a CMake option.
- If some level of isolation is desired , let your cmake project copy the downloaded file into the binary dir, and use that dir to access the file.
